pkg/appgw/configbuilder.go

```diff
@@ -17,6 +17,7 @@ type ConfigBuilder interface {
 	// builder pattern
 	BackendHTTPSettingsCollection(ingressList []*v1beta1.Ingress) error
 	BackendAddressPools(ingressList []*v1beta1.Ingress) error
+	NewBackendPoolMap() map[backendIdentifier](*network.ApplicationGatewayBackendAddressPool)
```

Comment:
@rheaj99 I don't think this method belongs to the interface itself.

Typically in `interface` you would place signatures of functions that become part of the API contract. In our case here - NewBackendPooMap is an internal implementation detail of how we generate an intermediate struct to help us get to the final JSON config.

Let's remove it from the interface declaration and we can make it lowercase (private) - `newBackendPoolMap`